docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/service.cs

```diff
@@ -0,0 +1,113 @@
+        [OperationContract]
+        double Multiply(double n1, double n2);
+        [OperationContract]
+        double Divide(double n1, double n2);
+    }
+
+    // Service class which implements the service contract.
+    // Added code to write output to the console window
+    public class CalculatorService : ICalculator
+    {
+        public double Add(double n1, double n2)
+        {
+            double result = n1 + n2;
+            Console.WriteLine("Received Add({0},{1}). Return: {2}", n1, n2, result);
+            return result;
+        }
+
+        public double Subtract(double n1, double n2)
+        {
+            double result = n1 - n2;
+            Console.WriteLine("Received Subtract({0},{1}). Return: {2}", n1, n2, result);
+            return result;
+        }
+
+        public double Multiply(double n1, double n2)
+        {
+            double result = n1 * n2;
+            Console.WriteLine("Received Multiply({0},{1}). Return: {2}", n1, n2, result);
+            return result;
+        }
+
+        public double Divide(double n1, double n2)
+        {
+            double result = n1 / n2;
+            Console.WriteLine("Received Divide({0},{1}). Return: {2}", n1, n2, result);
+            return result;
+        }
+
+
+        // Host the service within console application.
+        public static async Task Main()
+        {
+            var builder = WebApplication.CreateBuilder();
+
+            // Set up port (previously this was done in configuration,
+            // but CoreWCF requires it be done in code)
+            builder.WebHost.UseNetTcp(8090);
+            builder.WebHost.ConfigureKestrel(options =>
+            {
+                options.ListenAnyIP(8080);
+                
+            });
+
+            // Add CoreWCF services to the ASP.NET Core app's DI container
+            builder.Services.AddServiceModelServices()
+                            .AddServiceModelConfigurationManagerFile("wcf.config")
+                            .AddServiceModelMetadata()
+                            .AddTransient<CalculatorSample.CalculatorService>();
+
+            var app = builder.Build();
+
+            // Enable getting metadata/wsdl
+            var serviceMetadataBehavior = app.Services.GetRequiredService<ServiceMetadataBehavior>();
+            serviceMetadataBehavior.HttpGetEnabled = true;
+            serviceMetadataBehavior.HttpGetUrl = new Uri("http://localhost:8080/CalculatorSample/metadata");
+
+            // Configure CoreWCF endpoints in the ASP.NET Core hosts
+            app.UseServiceModel(serviceBuilder =>
+            {
+                serviceBuilder.AddService<CalculatorSample.CalculatorService>(serviceOptions => 
+                {
+                    serviceOptions.DebugBehavior.IncludeExceptionDetailInFaults = true;
+                });
+
+                serviceBuilder.ConfigureServiceHostBase<CalculatorSample.CalculatorService>(serviceHost =>
+                {
+
```

Comment:
Why is this empty? Is this call needed, it looks like it should be removed?

Reply:
Hi! This is part of the template code added to the original code after WCF update. In this particular case, the sample is very easy so there is not any code needed to configure the service host. This delegate is still remained here because the <httpPageGetEnabled> property under serviceDebug is defaulted to be true. When it's false and no more other serviceDebug properties are stated, this delegate will be removed. The other empty lines are also results from the update, and I think it might be helpful to keep them so that the user can see the exact same code after they upgraded this sample on their own.